it/uk/gov/hmrc/agentmapping/stubs/AuthStubs.scala

```diff
@@ -1,73 +1,112 @@
 package uk.gov.hmrc.agentmapping.stubs
 
 import com.github.tomakehurst.wiremock.client.WireMock._
-import uk.gov.hmrc.auth.core.{ AffinityGroup, Enrolment }
+import uk.gov.hmrc.auth.core.{AffinityGroup, Enrolment}
 
 trait AuthStubs {
 
-  def givenUserNotAuthorisedWithError(mdtpDetail: String): Unit = {
-    stubFor(post(urlEqualTo("/auth/authorise"))
-      .willReturn(aResponse()
-        .withStatus(401)
-        .withHeader("WWW-Authenticate", s"""MDTP detail="${mdtpDetail}"""")))
-  }
+  def givenUserNotAuthorisedWithError(mdtpDetail: String): Unit =
+    stubFor(
+      post(urlEqualTo("/auth/authorise"))
+        .willReturn(
+          aResponse()
+            .withStatus(401)
+            .withHeader("WWW-Authenticate", s"""MDTP detail="$mdtpDetail"""")))
 
-  def givenUserAuthorisedFor(enrolment: String, identifierName: String, identifierValue: String, ggCredId: String, affinityGroup: AffinityGroup = AffinityGroup.Agent, agentCodeOpt: Option[String]): Unit = {
-    stubFor(post(urlEqualTo("/auth/authorise")).atPriority(1)
-      .withRequestBody(
-        equalToJson(
+  def givenUserIsAuthorisedFor(
+    serviceName: String,
+    identifierName: String,
+    identifierValue: String,
+    ggCredId: String,
+    affinityGroup: AffinityGroup = AffinityGroup.Agent,
+    agentCodeOpt: Option[String]): Unit = {
+    stubFor(
+      post(urlEqualTo("/auth/authorise"))
+        .atPriority(1)
+        .withRequestBody(equalToJson(
           s"""{"authorise":[
              |  {"authProviders":["GovernmentGateway"]},
-             |  {"affinityGroup":"$affinityGroup"},
-             |  {
-             |    "identifiers":[
-             |      {"key":"$identifierName","value":"$identifierValue"}
-             |    ],
-             |    "state":"Activated",
-             |    "enrolment":"$enrolment"
-             |  }
+             |  {"affinityGroup":"$affinityGroup"}
              |],
-             |"retrieve":["credentials","agentCode"]
-          }""".stripMargin, true, false))
-      .willReturn(aResponse()
-        .withStatus(200)
-        .withHeader("Content-Type", "application/json")
-        .withBody(s"""{ "credentials": {"providerId": "$ggCredId", "providerType": "GovernmmentGateway"} ${agentCodeOpt.map(ac => s""", "agentCode": "$ac" """).getOrElse("")} }""")))
+             |"retrieve":["credentials","agentCode","allEnrolments"]
+          }""".stripMargin,
+          true,
+          false
+        ))
+        .willReturn(
+          aResponse()
+            .withStatus(200)
+            .withHeader("Content-Type", "application/json")
+            .withBody(s"""
+                         |{ "credentials": {
+                         |    "providerId": "$ggCredId",
+                         |    "providerType": "GovernmmentGateway"
```

Comment:
Doubt it affects anything in this repo, but `"GovernmmentGateway"` only needs one `m`